site/en/docs/devtools/recorder/index.md

```diff
@@ -142,3 +157,59 @@ There are options to add and remove steps too. This is useful if you want to add
     - count: **9** (click **add count** button)
    {% Img src="image/dPDCek3EhZgLQPGtEG3y0fTn4v82/zeK91O21jvc5OD9HuovR.png", alt="The new step for coffee checkout has been updated with the aforementioned details.", width="800", height="663" %}
 6. [Replay](#replay) the flow now to see the changes.
+
+## Understanding the recording's selector {: #selector }
+
+During recording, the **Recorder** automatically detects two types of selectors for most of the steps: ARIA and CSS.
+
+{% Aside %}
+For more information on ARIA selectors, see [Syntactic vs. semantic selectors](/blog/puppetaria/#syntactic-vs-semantic-selectors).
+{% endAside %}
+
+For simple webpages, `id` attributes and CSS `class` attributes are sufficient for the **Recorder** to detect the selectors. However, that might not always be the case.
+
+### Common test selectors {: common-test-selector}
+
+For applications developed with modern JavaScript frameworks, (for example, [React](https://reactjs.org/), [Angular](https://angular.io/), [Vue](https://vuejs.org/), and CSS frameworks), the CSS `class` values might be autogenerated. For example:
+
+{% Img src="image/NJdAV9UgKuN8AhoaPBquL7giZQo1/ZtK52PaMMzKWiiAcsQfH.png", alt="Autogenerated CSS classes with randomized names", width="800", height="654" %}
+
+There are already some common popular test selectors in the market and the **Recorder** supports them as well. If you have the following common test selectors defined, the **Recorder** will automatically detect and use them first:
+
+- `data-testid`
+- `data-test`
+- `data-qa`
+- `data-cy`
+- `data-test-id`
+- `data-qa-id`
+- `data-testing`
+
+To create more resilient tests, consider using these test selectors.
+
```

Comment:
Should we provide a screenshot or animated example here?

Reply:
What do you want to show here animated?

I think even a text example can do. I'll add it.